test/test_watch.rb

```diff
@@ -102,6 +102,42 @@ def test_watch_pod_max_redirect
     end
   end
 
+  def test_watch_pod_api_bearer_token_file_success
+    stub_core_api_list
+
+    file = Tempfile.new('token')
+    client = Kubeclient::Client.new(
+      'http://localhost:8080/api/', 'v1',
+      auth_options: { bearer_token_file: file.path }
+    )
+
+    begin
+      file.write("valid_token")
+      file.rewind
+      stub_token = stub_request(:get, %r{/watch/pods})
+        .with(headers: { Authorization: 'Bearer valid_token' })
+        .to_return(body: open_test_file('watch_stream.json'), status: 200)
+
+      got = nil
+      client.watch_pods(as: :raw).each { |notice| got = notice }
```

Comment:
Beautiful tests overall :clap:  

However, I've checked your branch locally, re-ordered the commits, and confirmed the new tests also pass before your fix.  
This one-liner creates a WatchStream and calls `.each` at same time.  
However, if you extract `watcher = client.watch_pods(as: :raw)` to the top, then call `watcher.each { ... }` after the file writes, it fails without the fix and passes with it.

Reply:
o wow, thanks for testing it. There are some nuances which I missed :sweat_smile:

Reply:
Should be better now